### PR TITLE
3.next - Add small and tiny integer types

### PR DESCRIPTION
**Pull request by markstory**
Add `smallint` and `tinyint` types to the schema system. I'm targeting this for 3.5 as I don't think column types changing is acceptable in a bugfix release.

Postgres doesn't support tinyint, so it is mapped to smallint.

Refs #10347

### CODE REVIEW

File: src/Database/Schema/SqlserverSchema.php [hunk 1]
```diff
@@ -88,18 +88,18 @@ protected function _convertColumn($col, $length = null, $precision = null, $scal
             return ['type' => 'timestamp', 'length' => null];
         }
 
+        if ($col === 'tinyint') {
+            return ['type' => 'tinyint', 'length' => $precision ?: 3];
+        }
+        if ($col === 'smallint') {
+            return ['type' => 'smallint', 'length' => $precision ?: 5];
```
Comment (inoas) [inline src/Database/Schema/SqlserverSchema.php]: 'smallinteger' like 'biginteger' and 'integer'?

File: src/Database/Schema/SqlserverSchema.php [hunk 2]
```diff
@@ -88,18 +88,18 @@ protected function _convertColumn($col, $length = null, $precision = null, $scal
             return ['type' => 'timestamp', 'length' => null];
         }
 
+        if ($col === 'tinyint') {
+            return ['type' => 'tinyint', 'length' => $precision ?: 3];
```
Comment (inoas) [inline src/Database/Schema/SqlserverSchema.php]: tinyinteger

**Review by inoas — COMMENTED** [review]
Internal type Naming seems to become inconsistent.

File: src/Database/Schema/SqlserverSchema.php (hunk 2 above)
Comment (markstory) [inline src/Database/Schema/SqlserverSchema.php]: That's fair. I'll change it.

Comment (sebastienbarre) [inline src/Database/Schema/SqlserverSchema.php]: Hmmm. I can see something coming my way in 2.x :)

Comment (markstory) [inline src/Database/Schema/SqlserverSchema.php]: Yes, I've opened a pull request against your fork to update it.

Comment (sebastienbarre) [inline src/Database/Schema/SqlserverSchema.php]: Reviewed, thanks. All good except for Sqlite.

**Comment by markstory** [comment]
Any other feedback on this before I merge it?

File: src/View/Helper/FormHelper.php [hunk 3]
```diff
@@ -84,10 +84,20 @@ class FormHelper extends Helper
...
         'errorClass' => 'form-error',
         'typeMap' => [
-            'string' => 'text', 'datetime' => 'datetime', 'boolean' => 'checkbox',
-            'timestamp' => 'datetime', 'text' => 'textarea', 'time' => 'time',
-            'date' => 'date', 'float' => 'number', 'integer' => 'number',
-            'decimal' => 'number', 'binary' => 'file', 'uuid' => 'string'
+            'string' => 'text',
+            'datetime' => 'datetime',
+            'boolean' => 'checkbox',
+            'timestamp' => 'datetime',
```
Comment (ravage84) [inline src/View/Helper/FormHelper.php]: Nitpicking move one up, next to ``datetime``.

File: src/View/Helper/FormHelper.php [hunk 4]
```diff
@@ -84,10 +84,20 @@ class FormHelper extends Helper
...
+            'text' => 'textarea',
```
Comment (ravage84) [inline src/View/Helper/FormHelper.php]: Nitpicking: Move up to ``string``.

File: src/View/Helper/FormHelper.php [hunk 5]
```diff
@@ -84,10 +84,20 @@ class FormHelper extends Helper
...
+            'time' => 'time',
+            'date' => 'date',
+            'float' => 'number',
+            'integer' => 'number',
+            'tinyinteger' => 'number',
+            'smallinteger' => 'number',
+            'decimal' => 'number',
```
Comment (ravage84) [inline src/View/Helper/FormHelper.php]: Nitpicking: Move up to ``float``.

**Review by ravage84 — COMMENTED** [review]
Reordering of the type map, would better group the related types together.

**Pull request merged**